src/zarr/abc/store.py

```diff
@@ -112,81 +72,55 @@ async def _open(self) -> None:
         ------
         ValueError
             If the store is already open.
-        FileExistsError
-            If ``mode='w-'`` and the store already exists.
-
-        Notes
-        -----
-        * When ``mode='w'`` and the store already exists, it will be cleared.
         """
         if self._is_open:
             raise ValueError("store is already open")
-        if self.mode.str == "w":
-            await self.clear()
-        elif self.mode.str == "w-" and not await self.empty():
-            raise FileExistsError("Store already exists")
         self._is_open = True
 
     async def _ensure_open(self) -> None:
         """Open the store if it is not already open."""
         if not self._is_open:
             await self._open()
 
-    @abstractmethod
-    async def empty(self) -> bool:
+    async def empty_dir(self, prefix: str = "") -> bool:
```

Comment:
Do we want to make the _default_ behavior of `empty_dir` to wipe the entire store at its root? Seems a bit dangerous to me. A safer alternative might be to make the `prefix` a required keyword argument.

Reply:
`empty_dir` only checks if a directory is empty (`delete_dir` does the deleting). But I'm happy to remove the default.

Reply:
huh, the name `empty_dir` made me think it was a method that would empty a directory, not check if one was empty. Do we think this has enough confusion potential to warrant a slightly different name?

Reply:
Changed to `is_empty(prefix: str)`